docs/understanding-airbyte/connections/README.md

```diff
@@ -1,19 +1,40 @@
 # Connections
 
-A connection is a configuration for syncing data between a source and a destination. To setup a connection, a user must configure things such such as:
+A connection is a configuration for syncing data between a source and a destination. To setup a connection, a user must configure things such as:
 
 * A catalog selection: which [streams and fields](../catalog.md) to replicate from the source
-* Sync mode: Whether streams should be replicated using:
-  * [Full Refresh Overwrite](full-refresh-overwrite.md): Sync the whole stream and replace data in destination by overwriting it.
-  * [Full Refresh Append](full-refresh-append.md): Sync the whole stream and append data in destination.
-  * [Incremental Append](incremental-append.md): Sync new records from stream and append data in destination.
-  * [Incremental Deduped History](incremental-deduped-history.md): Sync new records from stream and append data in destination, also provides a deduplicated view mirroring the state of the stream in the source.
+* Sync mode: how streams should be replicated (read and write):
 * Sync schedule: when to trigger a sync of the data
+* Optional transformations: how to convert Airbyte protocol messages (raw JSON blob) data into some other data representations. 
 
-Sync schedules are explained below. For information about catalog selections, see [AirbyteCatalog & ConfiguredAirbyteCatalog](../catalog.md).
+## Sync modes
+
+Note that Sync mode names are generally composed of two parts on purpose. This is because Source and Destination connectors are decoupled from each other.
+One is responsible for reading, and the other for writing.
+
+1. The first part of the sync mode name denotes how source connectors work:
+  
+  * Incremental: Read only the last records in the source since last read operation during a previous sync
+      * Method 1: comparing to a cursor value
+      * Method 2: using [CDC](../cdc.md), only available for certain limited number of sources that can support it.
+  * Full Refresh: Read everything in the source
+
+2. The second part of the sync mode name denotes how destination connectors work, regardless of how the source is producing data:
+
+  * Overwrite: Delete and then write over data in the destination that already exists
+  * Append: Write by adding data to existing tables in the destination
+  * Deduped History: Write by adding data to existing tables in the destination to keep a history of changes. The final table is produced by deduplicating the intermediate ones using a primary key.
+
+A sync mode is therefore, a combination of a source and destination mode together. The UI exposes the following options, whenever both source and destination connectors are capable to support it for the corresponding stream:
```

Comment:
this is great!

Reply:
+1

Reply:
@avaidyanatha could we add some illustrations about the sync modes? This can help users understand the concept easier.
![image](https://user-images.githubusercontent.com/5154322/120338780-ee2c6000-c2ca-11eb-9497-1c732bb3f7f2.png)